src/cargo/core/package_id.rs

```diff
@@ -252,15 +252,15 @@ mod tests {
         let loc = CRATES_IO_INDEX.into_url().unwrap();
         let pkg_id = PackageId::new("foo", "1.0.0", SourceId::for_registry(&loc).unwrap()).unwrap();
         assert_eq!(
-            r#"PackageId { name: "foo", version: "1.0.0", source: "registry `https://github.com/rust-lang/crates.io-index`" }"#,
+            r#"PackageId { name: "foo", version: "1.0.0", source: "registry `crates-io`" }"#,
             format!("{:?}", pkg_id)
         );
 
         let expected = r#"
 PackageId {
     name: "foo",
     version: "1.0.0",
-    source: "registry `https://github.com/rust-lang/crates.io-index`",
+    source: "registry `crates-io`",
```

Comment:
Why not keep the original registry url instead of using a short alias?

Reply:
It's the intent of #6691, which make it less mess in terminal (maybe?). And yes it gets some rooms for deciding whether URL should be hidden.

Reply:
I think it isn't an issue since it's less than 80 characters which terminal will usually expect to be the lowest, of course it can go lower but most stuff will look ugly.
```
    Updating `https://gitlab.com/xxxxxxxxxx/cargo/crates-index` index
```
If a short alias is used some amount of information is hidden which could be useful in the logs.

Reply:
Yep. Maybe we can preserve the original behaviour of crates.io, and shorten other custom registries, since we have no idea which contains sensitive information. Is this more reasonable?

Reply:
Is the url sensitive? If it is sensitive shouldn't the user hide it themselves? I think it is not easily missed as it is at the top line, and it does not contain any credentials.

Reply:
I got the point and totally agreed. It's very reasonable not using short alias. Let's await the opinion from others. Maybe we can come up with better solution.

Reply:
I'm a bit confused about the question here.  Are you just talking about the debug display?  That is almost never shown anywhere, so I'm not sure it really matters.  It'll probably be safer to keep the short name (like if asked to share `CARGO_LOG` output).  If I'm misunderstanding, can you clarify?

Reply:
Not just debug display, it affects the default cargo display as well. Rather than showing the crates url it shows `crates-io`.